### PR TITLE
Add system test and doc to http

### PR DESCRIPTION
**Pull request by feluelle**
Add http system test calling the pre-existing http example dag.
The example dag contains different HTTP operations. All requests will be called against the httpbin host.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by feluelle** [comment]
@turbaszek Is this fine or should I expose the params via env variables?

File: airflow/providers/http/example_dags/example_http.py
Comment (turbaszek) [inline airflow/providers/http/example_dags/example_http.py]:
```suggestion
post_op = SimpleHttpOperator(
    task_id='post_op',
```
I like it when task instance and task_id are inline.WDYT?

File: tests/providers/http/operators/test_http_system.py
Comment (turbaszek) [inline tests/providers/http/operators/test_http_system.py:31]: I'm wondering if we should use more generic system type like "core" or something. WDYT? @potiuk ?

**Review by turbaszek — APPROVED** [review]
Looks good, only two questions

**Comment by turbaszek** [comment]
> @turbaszek Is this fine or should I expose the params via env variables? When and which params should we make available to env variables?



> @turbaszek Is this fine or should I expose the params via env variables? When and which params should we make available to env variables?

All ids should be env variables (for security reasons). In case of GCP buckets names have to be unique (and constant during dag run so no way to generate them in code) that's why we use env variables.

File: docs/howto/operator/http/http.rst
Comment (kaxil) [inline docs/howto/operator/http/http.rst]:
```suggestion
The following code examples use the ``http_default`` connection which means the requests are sent against `httpbin <https://www.httpbin.org/>`__ site to perform basic HTTP operations.
```

File: docs/howto/operator/http/http.rst
Comment (kaxil) [inline docs/howto/operator/http/http.rst]:
```suggestion
In this example we call a ``DELETE`` operation to the ``delete`` endpoint. This time we are passing form data to the
```

File: docs/howto/operator/http/http.rst
Comment (kaxil) [inline docs/howto/operator/http/http.rst]:
```suggestion
Here we pass form data to a ``POST`` operation which is equal to a usual form submit.
```

**Review by kaxil — CHANGES_REQUESTED** [review]

File: docs/howto/operator/http/http.rst
Comment (kaxil) [inline docs/howto/operator/http/http.rst:24]: Thanks for adding docs, always love seeing additional docs

File: airflow/providers/http/example_dags/example_http.py
Comment (feluelle) [reply airflow/providers/http/example_dags/example_http.py]: me too - will change it :)

**Comment by feluelle** [comment]
@kaxil fixed the doc issues but wait for ci before you merge please 😁 ..did not test the docs. I also changed the task var names and I hope I refactored it correctly. (if there are issues I am gonna fix it tomorrow :))

File: tests/providers/http/operators/test_http_system.py
Comment (mik-laj) [reply tests/providers/http/operators/test_http_system.py:31]: It is not the core operator. It is in the provider directory. This allows it to be released independently from Airflow.  It is common and mature, but its use and usage are still developing. It is flexible enough to be extended with new authentication mechanisms and more. Last week we had a valuable and non-doc contribution to this operator.
https://github.com/apache/airflow/pull/8429

Here is list of all core operators.
```
airflow.operators.bash.BashOperator
airflow.operators.branch_operator.BaseBranchOperator
airflow.operators.check_operator.CheckOperator
airflow.operators.check_operator.IntervalCheckOperator
airflow.operators.check_operator.ThresholdCheckOperator
airflow.operators.check_operator.ValueCheckOperator
airflow.operators.dagrun_operator.TriggerDagRunOperator
airflow.operators.dummy_operator.DummyOperator
airflow.operators.generic_transfer.GenericTransfer
airflow.operators.latest_only_operator.LatestOnlyOperator
airflow.operators.python.BranchPythonOperator
airflow.operators.python.PythonOperator
airflow.operators.python.PythonVirtualenvOperator
airflow.operators.python.ShortCircuitOperator
airflow.operators.subdag_operator.SubDagOperator
```

**Comment by zhongjiajie** [comment]
> @kaxil fixed the doc issues but wait for ci before you merge please 😁 ..did not test the docs. I also changed the task var names and I hope I refactored it correctly. (if there are issues I am gonna fix it tomorrow :))

FYI, still failed 😢

**Comment by zhongjiajie** [comment]
I think it's failed due to
```log
2020-04-27T18:10:16.7233125Z [01mlooking for now-outdated files... [39;49;00mnone found
2020-04-27T18:10:16.9090056Z [01mpickling environment... [39;49;00mdone
2020-04-27T18:10:16.9090564Z [01mchecking consistency... [39;49;00m[91m/opt/airflow/docs/howto/operator/http/http.rst: WARNING: document isn't included in any toctree[39;49;00m
```

File: tests/providers/http/operators/test_http_system.py
Comment (feluelle) [reply tests/providers/http/operators/test_http_system.py:31]: Agree with Kamil 👍

**Review by turbaszek — APPROVED** [review]

**Review by zhongjiajie — APPROVED** [review]

**Comment by zhongjiajie** [comment]
FYI, I change the PR titile

File: docs/exts/exampleinclude.py
Comment (feluelle) [inline docs/exts/exampleinclude.py]: BTW this fixes a deprecation warning.

Comment (feluelle) [reply docs/exts/exampleinclude.py]: I added the info to the commit msg.

Comment (mik-laj) [reply docs/exts/exampleinclude.py]: Can you do it in a separate PR? We would like to have these changes in Airflow 1.10 too.

**Review by kaxil — APPROVED** [review]

File: docs/exts/exampleinclude.py
Comment (kaxil) [reply docs/exts/exampleinclude.py]: yeah would be good to have it in a separate PR

Comment (feluelle) [reply docs/exts/exampleinclude.py]: Good point - will do it separately.

**Comment by zhongjiajie** [comment]
Merging it, Thanks @feluelle

**Pull request merged**
